### PR TITLE
In d2iGenericKey(), if a falcon key is encountered, make a dummy pkey.

### PR DESCRIPTION
**Pull request by anhu**
This allows apache-httpd to work without PQ-specific patch along with a previous
pull request.

### CODE REVIEW

**Comment by anhu** [comment]
Tested against apache. "It works!"

File: wolfssl/openssl/evp.h [hunk 1]
```diff
@@ -247,6 +247,7 @@ enum {
     NID_rc4           = 5,
     EVP_PKEY_DH       = NID_dhKeyAgreement,
     EVP_PKEY_HMAC     = NID_hmac,
+    EVP_PKEY_FALCON = 300,
```
Comment (dgarske) [inline wolfssl/openssl/evp.h:250]: How'd you come up with 300?

File: src/ssl.c [hunk 2]
```diff
@@ -8522,6 +8522,70 @@ static WOLFSSL_EVP_PKEY* d2iGenericKey(WOLFSSL_EVP_PKEY** out,
...
+        }
+    #else
+        falcon_key falcon[1];
+    #endif
+        XMEMSET(falcon, 0, sizeof(falcon_key));
+
+        /* test if Falcon key */
+        if (priv) {
+            /* Try level 1 */
+            isFalcon = wc_falcon_init(falcon) == 0 &&
```
Comment (dgarske) [inline src/ssl.c:8542]: Maybe call `wc_falcon_init` only once at the start?

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfssl/openssl/evp.h (hunk 1 above)
Comment (anhu) [reply wolfssl/openssl/evp.h:250]: Pull it out of thin air.  Do you have a suggestion for something different?

File: src/ssl.c (hunk 2 above)
Comment (anhu) [reply src/ssl.c:8542]: Sure, I can do that.

File: wolfssl/openssl/evp.h (hunk 1 above)
Comment (dgarske) [reply wolfssl/openssl/evp.h:250]: No that is fine, but please comment as such. Usually these are NID values.

Comment (anhu) [reply wolfssl/openssl/evp.h:250]: Got it...will add a comment.

**Comment by anhu** [comment]
Just noticed the failed test.  Will get to it on Monday.

**Review by dgarske — APPROVED** [review]

**Pull request merged**